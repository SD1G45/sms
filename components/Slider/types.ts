import React from "react";

export interface SliderProps {
  value: number;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
