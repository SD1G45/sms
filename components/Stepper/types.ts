export interface StepperProps {
  steps: string[];
  activeIndex: number;
  className?: string;
}

export interface StepProps {
  active: boolean;
}
