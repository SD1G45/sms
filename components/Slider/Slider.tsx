import React from "react";
import styled from "styled-components";
import { SliderProps } from "./types";

const sliderThumbStyles = (props: any) => `
  width: 25px;
  height: 25px;
  background: ${props.theme.colors.primary};
  border-radius: 50%;
  cursor: pointer;
  -webkit-transition: .2s;
  transition: opacity .2s;
`;

const Styles = styled.div`
  display: flex;
  align-items: center;
  color: #888;
  margin-top: 2rem;
  margin-bottom: 2rem;
  .value {
    flex: 1;
    font-size: 2rem;
  }
  .slider {
    flex: 6;
    -webkit-appearance: none;
    width: 100%;
    height: 15px;
    border-radius: 5px;
    background: #efefef;
    outline: none;
    &::-webkit-slider-thumb {
      -webkit-appearance: none;
      appearance: none;
      ${(props) => sliderThumbStyles(props)}
    }
    &::-moz-range-thumb {
      ${(props) => sliderThumbStyles(props)}
    }
  }
`;

const Slider: React.FC<SliderProps> = ({ value, onChange }) => {
  return (
    <Styles>
      <input
        type="range"
        min={0}
        max={100}
        value={value}
        className="slider"
        onChange={onChange}
      />
    </Styles>
  );
};

export default Slider;
