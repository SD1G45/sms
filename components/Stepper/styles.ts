import styled from "styled-components";
import { StepProps } from "./types";

export const Container = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const Step = styled.button<StepProps>`
  width: 30px;
  height: 30px;
  border-radius: 50%;
  background-color: ${(props) =>
    props.active ? props.theme.colors.primary : "white"};
  color: ${(props) => (props.active ? "white" : props.theme.colors.primary)};
  border: ${(props) =>
    props.active ? "none" : "1px solid " + props.theme.colors.primary};
`;

export const Line = styled.div`
  flex-grow: 1;
  height: 2px;
  background-color: ${(props) => props.theme.colors.lightGray};
  margin: 0px 8px;
`;

export const StepTitle = styled.p`
  font-size: 16px;
  margin-left: 8px;
`;
