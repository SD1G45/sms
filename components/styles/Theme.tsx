import React from "react";
import { ThemeProvider } from "styled-components";
import ITheme from "./types";

const screenWidth = {
  minSmall: `(min-width: ${430}px)`,
  minMedium: `(min-width: ${730}px)`,
  minLarge: `(min-width: ${1024}px)`,
  maxSmall: `(max-width: ${430}px)`,
  maxMedium: `(max-width: ${730}px)`,
  maxLarge: `(max-width: ${1024}px)`,
};

const lightTheme: ITheme = {
  colors: {
    primary: "#4881F0",
    lightGray: "#BEC8D2",
    textPrimary: "#000000",
    error: "#FF7F7F",
  },
  screenWidth: { ...screenWidth },
};

interface ThemeProps {
  children: React.ReactNode;
}

const Theme: React.FC<ThemeProps> = ({ children }) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);

export default Theme;
