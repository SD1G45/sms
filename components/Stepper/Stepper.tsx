import React from "react";
import { Container, Line, Step, StepTitle } from "./styles";
import { StepperProps } from "./types";

const Stepper: React.FC<StepperProps> = ({ steps, activeIndex }) => {
  return (
    <Container>
      {steps.map((value, index) => (
        <>
          <Step active={activeIndex === index} key={index}>
            {index + 1}
          </Step>
          <StepTitle>{value}</StepTitle>
          {index !== steps.length - 1 && <Line />}
        </>
      ))}
    </Container>
  );
};

export default Stepper;
