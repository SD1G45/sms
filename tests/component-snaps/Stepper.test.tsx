import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import Slider from "../../components/Slider";
import Theme from "../../components/styles/Theme";
import Stepper from "../../components/Stepper";

const steps = ["Step 1", "Step 2", "Step 3"];

describe("<Slider/>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(
      <Theme>
        <Stepper steps={steps} activeIndex={1}/>
      </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});