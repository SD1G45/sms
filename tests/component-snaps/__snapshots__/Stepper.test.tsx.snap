// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider/> renders correctly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 30px;
  height: 30px;
  border-radius: 50%;
  background-color: white;
  color: #4881F0;
  border: 1px solid #4881F0;
}

.c4 {
  width: 30px;
  height: 30px;
  border-radius: 50%;
  background-color: #4881F0;
  color: white;
  border: none;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 2px;
  background-color: #BEC8D2;
  margin: 0px 8px;
}

.c2 {
  font-size: 16px;
  margin-left: 8px;
}

<div
  className="c0"
>
  <button
    className="c1"
  >
    1
  </button>
  <p
    className="c2"
  >
    Step 1
  </p>
  <div
    className="c3"
  />
  <button
    className="c4"
  >
    2
  </button>
  <p
    className="c2"
  >
    Step 2
  </p>
  <div
    className="c3"
  />
  <button
    className="c1"
  >
    3
  </button>
  <p
    className="c2"
  >
    Step 3
  </p>
</div>
`;
